src/components/PopupWithForm.js

```diff
@@ -26,4 +32,12 @@ export default class PopupWithForm extends Popup {
       evt.target.reset();
```

Comment:
Please, delete `evt.target.reset();` from `PopupWithForm` from line 32


You need to **reset** the form only  in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.